pinot-core/src/test/java/org/apache/pinot/queries/TransformQueriesTest.java

```diff
@@ -0,0 +1,286 @@
+
+    final MetricFieldSpec m3Spec = new MetricFieldSpec(M3, FieldSpec.DataType.LONG);
+    _schema.addField(m3Spec);
+
+    final MetricFieldSpec m4Spec = new MetricFieldSpec(M4, FieldSpec.DataType.LONG);
+    _schema.addField(m4Spec);
+
+    final TimeFieldSpec tSpec = new TimeFieldSpec(new TimeGranularitySpec(FieldSpec.DataType.LONG, TimeUnit.MILLISECONDS, TIME));
+    _schema.addField(tSpec);
+  }
+
+  @AfterClass
+  public void destroy() {
+    FileUtils.deleteQuietly(INDEX_DIR);
+  }
+
+  private void delete() {
+    for (IndexSegment indexSegment : _indexSegments) {
+      if (indexSegment != null) {
+        indexSegment.destroy();
+      }
+    }
+    _indexSegments.clear();
+  }
+
+  @Test
+  public void testTransformWithAvgInnerSegment() throws Exception {
+    try {
+      final List<GenericRow> rows = createDataSet(10);
+
+      final RecordReader recordReader = new GenericRowRecordReader(rows, _schema);
+      createSegment(_schema, recordReader, SEGMENT_NAME_1, TABLE_NAME);
+      final ImmutableSegment segment = loadSegment(SEGMENT_NAME_1);
+      _indexSegments.add(segment);
+
+      String query = "SELECT AVG(SUB(INT_COL1, INT_COL2)) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, -10000.00, 10);
+
+      query = "SELECT AVG(SUB(LONG_COL1, INT_COL1)) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, 4990000.00, 10);
+
+      query = "SELECT AVG(SUB(LONG_COL2, LONG_COL1)) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, 5000000.00, 10);
+
+      query = "SELECT AVG(ADD(INT_COL1, INT_COL2)) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, 30000.00,10);
+
+      query = "SELECT AVG(ADD(INT_COL1, LONG_COL1)) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, 5010000.00,10);
+
+      query = "SELECT AVG(ADD(LONG_COL1, LONG_COL2)) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, 15000000.00,10);
+
+      query = "SELECT AVG(ADD(DIV(INT_COL1, INT_COL2), DIV(LONG_COL1, LONG_COL2))) FROM foo";
+      runAndVerifyInnerSegmentQuery(query, 10.00, 10);
+
+      try {
+        query = "SELECT AVG(SUB(INT_COL1, STRING_COL)) FROM foo";
+        runAndVerifyInnerSegmentQuery(query, -10000.00, 10);
+        Assert.fail("Query should have failed");
+      } catch (Exception e) { }
+
+      try {
+        query = "SELECT AVG(ADD(DIV(INT_COL1, INT_COL2), DIV(LONG_COL1, STRING_COL))) FROM foo";
+        runAndVerifyInnerSegmentQuery(query, 10.00, 10);
+        Assert.fail("Query should have failed");
+      } catch (Exception e) { }
+    } finally {
+      delete();
+    }
+  }
+
+  @Test
+  public void testTransformWithAvgInterSegmentInterServer() throws Exception {
+    try {
+      final List<GenericRow> segmentOneRows = createDataSet(10);
+      final List<GenericRow> segmentTwoRows = createDataSet(10);
+
+      final RecordReader recordReaderOne = new GenericRowRecordReader(segmentOneRows, _schema);
+      final RecordReader recordReaderTwo = new GenericRowRecordReader(segmentTwoRows, _schema);
```

Comment:
Resource leak, use try-with resource

Reply:
Changed to try with resource but in this case we were not leaking resource as record reader.close() is a NO-OP